### PR TITLE
Assorted CI cleanup.

### PR DESCRIPTION
**Pull request by Ferroin**
##### Summary

This tidy’s up our CI, first fixing a number of issues reported by yamllint and actionlint, and then overhauling the CI failure notifications to provide significantly more useful information.

##### Component Name

area/ci

##### Test Plan

CI passes correctly on this PR.

### CODE REVIEW

**Review by iigorkarpov — APPROVED** [review]

File: .github/workflows/packaging.yml [hunk 1]
```diff
@@ -66,104 +66,106 @@ jobs:
...
-          - {distro: opensuse, version: "15.3", pkgclouddistro: opensuse/15.3, format: rpm, base_image: opensuse/leap, platform: linux/amd64, arch: amd64}
-          - {distro: opensuse, version: "15.3", pkgclouddistro: opensuse/15.3, format: rpm, base_image: opensuse/leap, platform: linux/arm64/v8, arch: arm64}
+          - {distro: opensuse, version: "15.2", pkgclouddistro: opensuse/15.2, format: rpm, base_image: opensuse/leap, platform: linux/amd64,
+             arch: amd64}
+          - {distro: opensuse, version: "15.2", pkgclouddistro: opensuse/15.2, format: rpm, base_image: opensuse/leap, platform: linux/arm64/v8,
+             arch: arm64}
+          - {distro: opensuse, version: "15.3", pkgclouddistro: opensuse/15.3, format: rpm, base_image: opensuse/leap, platform: linux/amd64,
+             arch: amd64}
+          - {distro: opensuse, version: "15.3", pkgclouddistro: opensuse/15.3, format: rpm, base_image: opensuse/leap, platform: linux/arm64/v8,
+             arch: arm64}
```
Comment (ilyam8) [inline .github/workflows/packaging.yml:76]: GitHub UI shows this as an error

<details>
<summary>unexpected end of the stream within a flow collection</summary>

Screenshot 2021-11-12 at 16 26 20

</details>

**Review by iigorkarpov — APPROVED** [review]

File: .github/workflows/packaging.yml (hunk 1 above)
Comment (ilyam8) [reply .github/workflows/packaging.yml]: not a problem I guess, it is inline dictionary, 🤷‍♂️ why it is being highlighted as an error

**Comment by Ferroin** [comment]
Rebased to resolve merge conflicts.

File: .github/workflows/packaging.yml (hunk 1 above)
Comment (Ferroin) [reply .github/workflows/packaging.yml]: And completely not a problem now since we’re generating this list at runtime from the consolidated CI matrix.

**Review by maneamarius — APPROVED** [review]

**Comment by Ferroin** [comment]
Rebased to fix merge conflicts and pick up CI changes.

**Pull request merged**